crates/ruff_linter/src/source_kind.rs

```diff
@@ -97,14 +99,31 @@ impl SourceKind {
         match (self, other) {
             (SourceKind::Python(src), SourceKind::Python(dst)) => {
                 let text_diff = TextDiff::from_lines(src, dst);
-                let mut unified_diff = text_diff.unified_diff();
-
-                if let Some(path) = path {
-                    unified_diff.header(&fs::relativize_path(path), &fs::relativize_path(path));
+                let unified_diff = text_diff.unified_diff();
+
+                let relativized_path = if let Some(path) = path {
+                    fs::relativize_path(path)
+                } else {
+                    String::new()
+                };
```

Comment:
This is not related to your change and might be worth a separate PR but showing the same header for both add and remove doesn't feel that useful... We might want to change this that the caller needs to provide labels.